cmd/asset-syncer/sync.go

```diff
@@ -34,36 +37,71 @@ var syncCmd = &cobra.Command{
 			return
 		}
 
-		mongoURL, err := cmd.Flags().GetString("mongo-url")
-		if err != nil {
-			logrus.Fatal(err)
-		}
-		mongoDB, err := cmd.Flags().GetString("mongo-database")
-		if err != nil {
-			logrus.Fatal(err)
-		}
-		mongoUser, err := cmd.Flags().GetString("mongo-user")
-		if err != nil {
-			logrus.Fatal(err)
-		}
-		mongoPW := os.Getenv("MONGO_PASSWORD")
 		debug, err := cmd.Flags().GetBool("debug")
 		if err != nil {
 			logrus.Fatal(err)
 		}
 		if debug {
 			logrus.SetLevel(logrus.DebugLevel)
 		}
-		mongoConfig := datastore.Config{URL: mongoURL, Database: mongoDB, Username: mongoUser, Password: mongoPW}
-		dbSession, err := datastore.NewSession(mongoConfig)
-		if err != nil {
-			logrus.Fatalf("Can't connect to mongoDB: %v", err)
+		database, err := cmd.Flags().GetString("database-type")
+		var manager assetManager
+		if database == "mongodb" {
```

Comment:
Same comment as previous PR - let's avoid sprinkling the implementation details throughout, and instead encapsulate the specifics in each interface implementation (and instantiate those in the main.go)